planner/core/find_best_task.go

```diff
@@ -571,8 +577,12 @@ func (is *PhysicalIndexScan) initSchema(id int, idx *model.IndexInfo, isDoubleRe
 	is.SetSchema(expression.NewSchema(indexCols...))
 }
 
-func (is *PhysicalIndexScan) addPushedDownSelection(copTask *copTask, p *DataSource, path *accessPath, finalStats *property.StatsInfo) {
+func (ds *DataSource) addPushedDownSelection(copTask *copTask, path *accessPath) error {
 	// Add filter condition to table plan now.
+	is, ok := copTask.indexPlan.(*PhysicalIndexScan)
+	if !ok {
+		return errors.Trace(fmt.Errorf("type assertion fail, expect PhysicalIndexScan, but got %v", reflect.TypeOf(copTask.indexPlan)))
```

Comment:
You can use `pingcap/errors.Errorf()`, which can generate a call stack.
```suggestion
		return errors.Errorf("type assertion failed, expect PhysicalIndexScan, but got %T", copTask.indexPlan)
```

Reply:
Fixed, PTAL~